protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```
@@ -331,7 +340,17 @@ impl ChannelFactory {
         let hom_group_id = 0;
         let mut result = vec![];
         let channel_id = id;
-        let target = crate::utils::hash_rate_to_target(downstream_hash_rate, self.share_per_min);
+        let target =
```

function should return an error not an option